### ISSUE TITLE
strange image artifact in training and testing 

### ISSUE THREAD

**Issue by crypticsymmetry** [NVIDIA/pix2pixHD#140]
not sure whats happening here. i get these strange artifacts that appear during training and testing. i have tried changing learning rate with no effect. Any help would be much appreciated! Thanks.

![output_0016_synthesized_image](https://user-images.githubusercontent.com/29524706/60833781-ec57d180-a173-11e9-9a2d-1550899fd1bf.jpg)
![output_0011_synthesized_image](https://user-images.githubusercontent.com/29524706/60833792-efeb5880-a173-11e9-8cab-d58f34043b56.jpg)

**Comment by nile649** [comment]
Gans is unstable

**Comment by Quasimondo** [comment]
Same problem as https://github.com/NVIDIA/pix2pixHD/issues/46 - it's a problem that haunts me since years - also pix2pix has it. I still have not found the cause, but once a model has caught these types of artifacts there seems to be no way to get rid of them, so you might want to restart the training again from scratch.

**Comment by Zephyr69** [comment]
I'm positive this kind of artifacts are caused by nn.Tanh that produces the final generator output. Maybe it is too complicated for the model to learn to utilize the full range of [-1, 1]. Or there are some misalignments in the input normalization methods used in this repo.

Switching to nn.Sigmoid completely eliminates these artifacts for me. Of course, you need to modify some input normalization methods to accomodate this change.

**Comment by zhangxiaojuan66** [comment]
你好，邮件已收到，祝你万事如意，生活愉快！

**Comment by chadify007** [comment]
> Switching to nn.Sigmoid completely eliminates these artifacts for me.

Interesting, so it isn't related to learning rate. I think instancenorm has some effect on this since it is looking at all the channels for each individual image which may lead to instability at lowest or highest luminosity regions?  Idk has anyone touched this lately?

**Comment by chadify007** [comment]
You can definitely tell it's hanging out in this case near the brightest portion of the image

**Comment by zhangxiaojuan66** [comment]
你好，邮件已收到，祝你万事如意，生活愉快！

**Comment by chadify007** [comment]
> Same problem as https://github.com/NVIDIA/pix2pixHD/issues/46 - it's a problem that haunts me since years - also pix2pix has it. I still have not found the cause, but once a model has caught these types of artifacts there seems to be no way to get rid of them, so you might want to restart the training again from scratch.

Such a bit issue for me, my models visually and stylewise are approaching the quality I desire yet, I cannot seem to find a way to remove these dark or bright (with multiple colours defects). Depending on whether your dataset is not "normalized or something??" And it's bright or dark biased, it'll cause these artifacts to hide or be attracted to such bright or dark areas in the images. I bet instancenorm is behind this to a degree, but still I prefer it over batchnorm or groupnorm which tends to be too slow to learn. Batch size of 1 and instancenorm leads to bigger and better transformations between images.  You can see it's attracted to his white collar here!  ![image](https://github.com/user-attachments/assets/85d538af-8894-4b8f-b0df-70da3463cf22)
